File: Array 101/Greatest_element_Right_side1.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int last=-1;
        int temp=0;
        for(int i=arr.size()-1;i>=0;i--)
        {
            temp=arr[i];
            arr[i]=last;
            if(last<temp)
                last=temp;
        }
        return arr;
    }
};

File: Array 101/Greatest_element_Right_side1.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        last=-1;
        temp=0;
        i=(len(arr)-1)
        while(i>=0):
            temp=arr[i]
            arr[i]=last
            if last<temp:
                last=temp
            i-=1
        return arr